src/NITTA/Model/Networks/Bus.hs

```diff
@@ -298,24 +302,80 @@ instance {-# OVERLAPS #-} ByTime (BusNetwork tag v x t) t where
     BindProblem (BusNetwork tag v x t) tag v x
     where
-    bindOptions BusNetwork{bnRemains, bnPus} = concatMap optionsFor bnRemains
+    bindOptions BusNetwork{bnRemains, bnPus} = if not $ null singleOptionsList then groupBinding singleOptionsList ++ concat singleOptionsList else []
         where
             optionsFor f =
                 [ Bind f puTitle
                 | (puTitle, pu) <- M.assocs bnPus
                 , allowToProcess f pu
                 ]
 
+            singleOptionsList = map optionsFor bnRemains
+
+            puFunctionDict = M.fromAscListWith (++) $
+                concatMap (\f -> [ (unitType pu, [(f, puTitle)])
+                | (puTitle, pu) <- M.assocs bnPus
+                , allowToProcess f pu
+                ]) bnRemains
+
+            puOptionsOne option= (L.nub fsts, L.nub snds)
+              where
+                fsts = map fst option
+                snds = map snd option
+
+            puOptions = map puOptionsOne $ M.elems puFunctionDict
+
+            prod a b = [ zip a s | s <- sequence $ replicate (length a) b]
+
+            toCountDict lst = M.fromList $ L.nub $ map (\x -> (x, count lst x)) lst
+                where
+                    count lst_ x = length $ filter (==x) lst_
+            fromCountDict dict = concatMap (\(value, count) -> replicate count value) $ M.toList dict
+
+            createDataMap elements = M.fromList $ map (\el -> (toCountDict $ map snd el, map fst el)) elements
+
+            listsOfValues = concatMap (uncurry prod) puOptions
+
+            afterFiltering = map (\(k, v) -> map (uncurry Bind ) $ zip v $ fromCountDict k )$ M.toList $ createDataMap listsOfValues
+
+            groupBinding options
+                | not $ null $ filter (\x -> length x > 1) options = trace (show $ M.elems puFunctionDict) $ map (GroupBinding AllBinds) $ afterFiltering
+                | otherwise = [GroupBinding NonAlternativeBinds $ concat options]
+
     bindDecision bn@BusNetwork{bnProcess, bnPus, bnBinded, bnRemains} (Bind f tag) =
         bn
             { bnPus = M.adjust (bind f) tag bnPus
             , bnBinded = registerBinding tag f bnBinded
             , bnProcess = execScheduleWithProcess bn bnProcess $ scheduleFunctionBind f
             , bnRemains = filter (/= f) bnRemains
             }
+    bindDecision n@BusNetwork{bnProcess, bnPus, bnBinded, bnRemains} gp@(GroupBinding NonAlternativeBinds binds) =
+        n
+            { -- { bnPus = bnPus'
+              -- , bnBinded = bnBinded'
+              bnPus = L.foldl' (\m (Bind f tag) -> M.adjust (bind f) tag m) bnPus binds
+            , bnBinded = L.foldl' (\m (Bind f tag) -> registerBinding tag f m) bnBinded binds
+            , bnProcess = execScheduleWithProcess n bnProcess $ scheduleGroupBinding gp
+            , bnRemains = bnRemains L.\\ map (\(Bind f _) -> f) binds
+            }
+
+    bindDecision n@BusNetwork{bnProcess, bnPus, bnBinded, bnRemains} gp@(GroupBinding AllBinds binds) =
```

Comment:
Maybe, here will be better something like: `foldl' (\n' b -> bindDecision n' b) n b`